### PR TITLE
[ML] add support for xlm_roberta tokenized models

### PR DESCRIPTION
**Pull request by benwtrent**
Many multi-lingual and newer models use a tokenization scheme similar to sentence-piece. This PR adds support for one of those tokenization schemes, XLMRoBERTa. 

The main changes are:
 - Support for xlm_roberta tokenization configuration
 - Adding `scores` to the vocabulary document stored, requiring that scores be the same size as the vocabulary
 - Adding a new flat text file to resources that is the spm char normalizer.

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by davidkyle** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by szabosteve — APPROVED** [review]
Thanks for writing the docs for these changes! LGTM!
I've added `xlm_roberta` to the list of tokenization values via [f0748b8](https://github.com/elastic/elasticsearch/pull/94089/commits/f0748b890149b2eb1a32c2b2e25eae17dbaaf476), hope you don't mind.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelVocabularyAction.java [hunk 1]
```diff
@@ -71,13 +75,22 @@ public Request(StreamInput in) throws IOException {
             } else {
                 this.merges = List.of();
             }
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_9_0)) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelVocabularyAction.java:78]: In 8.9 `TransportVersion` is an incrementing number, see https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/TransportVersion.java#L34

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelVocabularyAction.java [hunk 2]
```diff
@@ -71,13 +75,22 @@ public Request(StreamInput in) throws IOException {
...
+            }
         }
 
         @Override
         public ActionRequestValidationException validate() {
             ActionRequestValidationException validationException = null;
             if (vocabulary.isEmpty()) {
                 validationException = addValidationError("[vocabulary] must not be empty", validationException);
+            } else {
+                if (scores.isEmpty() == false && scores.size() != vocabulary.size()) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelVocabularyAction.java:91]:
```suggestion
                if (scores.size() != vocabulary.size()) {
```

**Review by davidkyle — APPROVED** [review]
LGTM

Can you look at the transport version number please

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**